src/pip/_internal/req/req_uninstall.py

```diff
@@ -46,6 +46,7 @@ def _script_names(dist, script_name, is_gui):
         bin_dir = bin_user
     else:
         bin_dir = bin_py
+    assert bin_dir is not None
```

Comment:
These two asserts can be removed if you annotate `_get_bin_user()` with `() -> str`